### ISSUE TITLE
Instance segmentation

### ISSUE THREAD

**Issue by xyclone10** [apple/ml-hypersim#74]
Hello, I want to ask regarding the dataset of instance segmentation.

Currently, my output of the model will be `(batch_size, num_labels, height, width)`, which each pixel will be represented by a label instead of using bounding box. Could you help me to explain how to use it? or is it actually possible to use in my case? Because I am a little bit confused to rearrange the classes. Any help would be good. Thanks a lot :)

**Comment by mikeroberts3000** [comment]
I'm sorry, I have no idea what you're asking.

**Comment by xyclone10** [comment]
Hello.

I apologize for being unclear in my question and thank you for your suggestions.

Currently, I am doing a depth estimation task that requires some instance segmentation tasks too, which is why I am considering Hypersim as my main dataset right now. The pre-trained model that I tried is a DINOv2 encoder and a simple linear layer for the decoder for now. Even though instance segmentation is still not in the DINOv2 downstream tasks, I want to try the instance segmentation task. Rather than the previous question that I asked, I will simplify my main question. **How to create the instance segmentation training data and labels within this Hypersim dataset?**

1. This is because I am still new to instance segmentation. So far to my knowledge, we need the bounding box for each instance, which if I am correct, I found in this file, `metadata_semantic_instance_bounding_box_object_aligned_2d_positions.hdf5` which the shape is `(number of instances, XYZ coordinate (in camera space))` for each scene, am I correct?
2. Next, I found the `mesh_objects_sii.hdf5` as described, which contained of semantic instance ID for each object ID. I think this one correlated to the object names listed in `metadata_objects.csv` because the shape is the same. Is this one for each scene too?
3. For each frame, the file named `frame.IIII.semantic_instance.hdf5` would be the one that contained semantic instance IDs. If I am correct, these IDs are the ones that being the values in the list of the `mesh_objects_sii.hdf5` file.

If 1 and 2 are for each scene, the one thing that I am confused about is that their shape is different, I haven't understood yet which one is the one that can be called "number of instances". Please feel free to ask for details if you are still can't understand my question. I would be grateful if you could answer these in an easier explanation to understand.

Thank you!

**Comment by mikeroberts3000** [comment]
Your revised question is much more clear. Thank you. I'm still confused though. Why can't you use our instance segmentation images? `frame.IIII.semantic_instance.hdf5` is an image that contains instance IDs. Why do you need to examine all of these other data files?

**Comment by mikeroberts3000** [comment]
Our semantic instance IDs are stable across images and camera trajectories within each scene. If you are restricting yourself to a single scene, and you see a semantic instance ID of _S_ in two images, then you know for certain that those pixels belong to the same semantic instance. But our semantic instance IDs are not globally stable across the entire dataset. If you see a semantic instance ID of _S_ in two images from two different scenes, this is a coincidence, and there is no relationship whatsoever between the two instances. They are not the same instance.

You can verify this yourself by looking at consecutive images within a scene. You will see that objects have the same IDs in both images. But when you look at images from different scenes, two totally different objects can have the same semantic ID. So you'll need to account for this in how you're training and validating your model.

It is also important to remember that the actual integer value of each instance ID is irrelevant, in the sense that you could shuffle all the instance IDs however you want, and you would obtain an equally valid, equally correct semantic instance segmentation. So your code needs to be permutation-invariant with respect to the semantic instance IDs. You might or might not be doing this already, but I wanted to mention it just in case.

**Comment by xyclone10** [comment]
I think I already got my answer then, I appreciate a lot of your great explanation!

Thank you!
